CMakeModules/CPackConfig.cmake

```diff
@@ -131,79 +131,69 @@ cpack_add_install_type(Extra DISPLAY_NAME "Extra")
   INSTALL_TYPES Development Runtime)
 
-if (PACKAGE_MKL_DEPS AND PACKAGE_GFX_DEPS)
-  cpack_add_component(cpu
-    DISPLAY_NAME "CPU Backend"
-    DESCRIPTION "This Backend allows you to run ArrayFire code on native CPUs."
-    DEPENDS mkl_dependencies gfx_dependencies
-    INSTALL_TYPES Development Runtime)
-  cpack_add_component(opencl
-    DISPLAY_NAME "OpenCL Backend"
-    DESCRIPTION "This Backend allows you to take advantage of OpenCL capable GPUs to run ArrayFire code. Currently ArrayFire does not support OpenCL for the Intel CPU on OSX."
-    DEPENDS mkl_dependencies gfx_dependencies
-    INSTALL_TYPES Development Runtime)
-  cpack_add_component(cuda
-    DISPLAY_NAME "CUDA Backend"
-    DESCRIPTION "This Backend allows you to take advantage of the CUDA enabled GPUs to run ArrayFire code. Please make sure you have CUDA toolkit installed or install CUDA dependencies component."
-    DEPENDS gfx_dependencies cuda_dependencies
-    INSTALL_TYPES Development Runtime)
-elseif (PACKAGE_MKL_DEPS)
+cpack_add_component(cuda
+  DISPLAY_NAME "CUDA Backend"
+  DESCRIPTION "This Backend allows you to take advantage of the CUDA enabled GPUs to run ArrayFire code. Please make sure you have CUDA toolkit installed or install CUDA dependencies component."
+  DEPENDS common_backend_dependencies cuda_dependencies
+  INSTALL_TYPES Development Runtime)
+
+if (PACKAGE_MKL_DEPS)
   cpack_add_component(cpu
     DISPLAY_NAME "CPU Backend"
     DESCRIPTION "This Backend allows you to run ArrayFire code on native CPUs."
-    DEPENDS mkl_dependencies
-    INSTALL_TYPES Development Runtime)
-  cpack_add_component(opencl
-    DISPLAY_NAME "OpenCL Backend"
-    DESCRIPTION "This Backend allows you to take advantage of OpenCL capable GPUs to run ArrayFire code. Currently ArrayFire does not support OpenCL for the Intel CPU on OSX."
-    DEPENDS mkl_dependencies
+    DEPENDS mkl_dependencies common_backend_dependencies
     INSTALL_TYPES Development Runtime)
-  cpack_add_component(cuda
-    DISPLAY_NAME "CUDA Backend"
-    DESCRIPTION "This Backend allows you to take advantage of the CUDA enabled GPUs to run ArrayFire code. Please make sure you have CUDA toolkit installed or install CUDA dependencies component."
-    DEPENDS cuda_dependencies
-    INSTALL_TYPES Development Runtime)
-elseif (PACKAGE_GFX_DEPS)
+  if (NOT APPLE)
+    cpack_add_component(opencl
+      DISPLAY_NAME "OpenCL Backend"
+      DESCRIPTION "This Backend allows you to take advantage of OpenCL capable GPUs to run ArrayFire code. Currently ArrayFire does not support OpenCL for the Intel CPU on OSX."
+      DEPENDS mkl_dependencies common_backend_dependencies opencl_dependencies
+      INSTALL_TYPES Development Runtime)
+  else ()
+    cpack_add_component(opencl
+      DISPLAY_NAME "OpenCL Backend"
+      DESCRIPTION "This Backend allows you to take advantage of OpenCL capable GPUs to run ArrayFire code. Currently ArrayFire does not support OpenCL for the Intel CPU on OSX."
```

Comment:
if we are going to have a separate section we should remove the Intel CPU line from here. Also you should mention CPUs and GPUs